### DISCUSSION TITLE
Accessing environment variable

### DISCUSSION THREAD

**Discussion by lzap** [community/community#26765] [Actions]
<p>Hello, I have defined a variable and I want to access it from a if statement:</p>
<pre><code class="lang-yaml">---
name: deploy
on:
  push:
    branches:
      - master
env:
  ALLOWED_OWNER: theforeman
  
publish:
  if: github.repository_owner == ${{ env.ALLOWED_OWNER }}
  ...
</code></pre>
<p>But I am getting: The workflow is not valid. .github/workflows/deploy.yml (Line: 87, Col: 9): Unrecognized named-value: ‘env’. Located at position 1 within expression: env.ALLOWED_OWNER</p>
<p>What is wrong? Thanks</p>

**Comment by Simran-B** [comment, marked as answer]
<p>The if condition is syntactically wrong, the entire expression must be wrapped in <code>${{ }}</code> or not at all.</p>
<p>But that aside, you cannot reference the <code>env</code> context here:</p>
<blockquote>
<p>You can use the <code>env</code> context in the value of any key in a <strong>step</strong> except for the <code>id</code> and <code>uses</code> keys.</p>
</blockquote>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/actions/learn-github-actions/contexts#env-context">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/actions/learn-github-actions/contexts#env-context" target="_blank" rel="noopener nofollow ugc">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/7819991/181099798-d326f8bc-daf6-426b-a75e-647f1287aab9.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/actions/learn-github-actions/contexts#env-context" target="_blank" rel="noopener nofollow ugc">Contexts - GitHub Docs</a></h3>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by lzap** [comment]
<p>Damn, thanks.</p>
<p>I need to run two of our jobs on the same conditions, it is a complex query unfortunately and I want to share it somewhere.</p>

**Comment by anomiex** [comment]
<aside class="quote no-group" data-username="lzap" data-post="3" data-topic="211159">
<div class="title">
<div class="quote-controls"></div>
 lzap:</div>
<blockquote>
<p>I need to run two of our jobs on the same conditions, it is a complex query unfortunately and I want to share it somewhere.</p>
</blockquote>
</aside>
<p>Your best bet for something like that is probably to have a job that calculates the condition as an output, then the other jobs <code>needs</code> the first one and access the output via <code>needs.first-job.outputs.whatever</code>.</p>
